src/core/domain/useCases/validators/NewResourceValidator.ts

```diff
@@ -0,0 +1,6 @@
+import { ResourceValidationError } from './errors/ResourceValidationError';
+
+export interface NewResourceValidator {
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  validate(...args: any[]): Promise<void | ResourceValidationError>;
+}
```

Comment:
If this is a generic interface, you can avoid the any type by using the T placeholder that can be specified when you implement the interface `NewResourceValidator<NewDataset>`
```suggestion
export interface NewResourceValidator<T> {
  validate(...args: T[]): Promise<void | ResourceValidationError>;
}
```

Reply:
I initially approached it precisely as you describe, since in the initial design I only considered the NewDataset parameter in the validator.

In a later step, as I also needed the MetadataBlocks, I had to remove the generic type, since the validator receives two parameters of different types: `resource: NewDataset, metadataBlocks: MetadataBlock[]`

Furthermore, it is possible that future validators will require different number parameters to carry out validations, so I find it appropriate to leave it open (with `any[]`) for the moment, just as we do with the UseCase base class params.